app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeViewModel.kt

```diff
@@ -121,22 +121,36 @@ class ComposeViewModel @Inject constructor(
         }
     }
 
-    private suspend fun addMediaToQueue(
+    suspend fun addMediaToQueue(
         type: QueuedMedia.Type,
         uri: Uri,
         mediaSize: Long,
-        description: String? = null
+        description: String? = null,
+        replaceItem: QueuedMedia? = null
     ): QueuedMedia {
-        val mediaItem = media.updateAndGet { mediaValue ->
+        var stashMediaItem: QueuedMedia? = null
```

Comment:
This is needed because the QueuedMedia can now be inserted in the middle of the list, so we can no longer use last() to fetch the newly added item.